ui/selectors/selectors.js

```diff
@@ -718,6 +717,60 @@ export function getSnapRegistryData(state, snapId) {
   return snapsRegistryData ? snapsRegistryData[snapId] : null;
 }
 
+/**
+ * Find and return Snap's latest version available in registry.
+ *
+ * @param state - Redux state object.
+ * @param snapId - ID of a Snap.
+ * @returns String SemVer version.
+ */
+export function getSnapLatestVersion(state, snapId) {
+  const snapRegistryData = getSnapRegistryData(state, snapId);
```

Comment:
I wonder if we shouldn't be creating reselect selectors here instead?

Reply:
Refactored everything to use memoized selectors with `createSelector`. I'm just not sure if using `useSelector` to pass the state to it is the best approach, but I do not see other way for passing the state in some established way from these components (let me know if you think there is something better).